apps/bajour/pages/search/index.tsx

```diff
@@ -0,0 +1,159 @@
+    }
+  `}
+`
+
+export const SearchPageWrapper = styled('div')`
+  padding-top: ${({theme}) => theme.spacing(6)};
+`
+
+export const SearchStatus = styled('div')`
+  margin: 0 auto;
+  padding: ${({theme}) => theme.spacing(4)};
+  text-align: center;
+`
+
+const ITEMS_PER_PAGE = 5
+
+const SearchPage: React.FC = () => {
+  const router = useRouter()
+  const {query} = router
+  const [rawQuery, setRawQuery] = useState((query.q as string) || '')
+  const [phraseQuery, setPhraseQuery] = useState((query.q as string) || '')
+  const [page, setPage] = useState(1)
+
+  const {
+    data: phraseData,
+    loading,
+    error
+  } = ApiV1.usePhraseQuery({
+    variables: {
+      query: phraseQuery,
+      take: ITEMS_PER_PAGE,
+      skip: (page - 1) * ITEMS_PER_PAGE
+    }
+  })
+
+  useEffect(() => {
+    if (query.q) {
+      setPhraseQuery(query.q as string)
+    }
+  }, [query.q])
+
+  const handleSearch = (e: React.FormEvent) => {
+    e.preventDefault()
+    router.push({
+      pathname: '/search',
+      query: {q: rawQuery}
+    })
+    setPhraseQuery(rawQuery)
+    setPage(1)
+  }
+
+  const handlePageChange = (event: React.ChangeEvent<unknown>, value: number) => {
+    setPage(value)
+  }
+
+  const articlesNodes =
+    (phraseData?.phrase && phraseData.phrase.articles && phraseData.phrase.articles.nodes) || []
+  const pagesNodes =
+    (phraseData?.phrase && phraseData.phrase.pages && phraseData.phrase.pages.nodes) || []
+
+  const totalArticlesCount = phraseData?.phrase?.articles?.totalCount || 0
+  const totalPagesCount = phraseData?.phrase?.pages?.totalCount || 0
+  const totalCount = totalArticlesCount + totalPagesCount
+
+  const modifiedArticlesNodes = articlesNodes.map(node => ({
+    __typename: 'ArticleTeaser',
+    article: node
+  }))
+
+  const modifiedPagesNodes = pagesNodes.map(node => ({
+    __typename: 'PageTeaser',
+    page: node
+  }))
+
+  const phraseResultTeasers = [...modifiedArticlesNodes, ...modifiedPagesNodes]
+
+  return (
+    <SearchPageWrapper>
+      <SearchForm onSubmit={handleSearch}>
+        <SearchInput type="text" label="Artikelsuche" onChange={e => setRawQuery(e.target.value)} />
+        <SearchButton onClick={handleSearch} aria-label="Artikelsuche">
+          <MdSearch size={28} />
+        </SearchButton>
+      </SearchForm>
+
+      {loading && <SearchStatus>Loading...</SearchStatus>}
+      {error && <SearchStatus>Error: {error.message}</SearchStatus>}
+      {phraseResultTeasers.length > 0 && (
+        <>
+          <NavbarPhraseResults teasers={phraseResultTeasers} />
+          {totalCount > ITEMS_PER_PAGE && (
+            <Pagination
+              count={Math.ceil(totalCount / ITEMS_PER_PAGE)}
+              page={page}
+              onChange={handlePageChange}
+              style={{marginTop: '20px'}}
+            />
+          )}
+        </>
+      )}
+    </SearchPageWrapper>
+  )
+}
+
+export const NavbarPhraseResults = ({teasers}: {teasers: any[]}) => {
```

Comment:
Why any?

Reply:
I tried using AricleList[] | PageList[] and making it strongly typed but there were weird typescript issues and after some time I got tired of them and used `any`

Reply:
```diff
diff --git a/apps/bajour/pages/search/index.tsx b/apps/bajour/pages/search/index.tsx
index d0e3599d..f31ccf97 100644
--- a/apps/bajour/pages/search/index.tsx
+++ b/apps/bajour/pages/search/index.tsx
@@ -1,6 +1,6 @@
 import {css, Pagination, styled, TextField} from '@mui/material'
 import {Button} from '@wepublish/ui'
-import {alignmentForTeaserBlock, ApiV1, Teaser} from '@wepublish/website'
+import {alignmentForTeaserBlock, ApiV1, articleToTeaser, Teaser} from '@wepublish/website'
 import {useRouter} from 'next/router'
 import {useEffect, useState} from 'react'
 import {MdSearch} from 'react-icons/md'
@@ -53,6 +53,16 @@ export const SearchPagination = styled(Pagination)`
   margin-top: 20px;
 `
 
+export const pageToTeaser = (page: ApiV1.Page): ApiV1.PageTeaser => ({
+  __typename: 'PageTeaser',
+  style: ApiV1.TeaserStyle.Default,
+  page,
+  image: null,
+  lead: null,
+  preTitle: null,
+  title: null
+})
+
 const ITEMS_PER_PAGE = 5
 
 const SearchPage: React.FC = () => {
@@ -103,15 +113,8 @@ const SearchPage: React.FC = () => {
   const totalPagesCount = phraseData?.phrase?.pages?.totalCount || 0
   const totalCount = totalArticlesCount + totalPagesCount
 
-  const modifiedArticlesNodes = articlesNodes.map(node => ({
-    __typename: 'ArticleTeaser',
-    article: node
-  }))
-
-  const modifiedPagesNodes = pagesNodes.map(node => ({
-    __typename: 'PageTeaser',
-    page: node
-  }))
+  const modifiedArticlesNodes = articlesNodes.map(node => articleToTeaser(node as ApiV1.Article))
+  const modifiedPagesNodes = pagesNodes.map(node => pageToTeaser(node as ApiV1.Page))
 
   const phraseResultTeasers = [...modifiedArticlesNodes, ...modifiedPagesNodes]
 
@@ -142,7 +145,7 @@ const SearchPage: React.FC = () => {
   )
 }
 
-export const NavbarPhraseResults = ({teasers}: {teasers: any[]}) => {
+export const NavbarPhraseResults = ({teasers}: {teasers: ApiV1.Teaser[]}) => {
   return (
     <NavbarPhraseResultsWrapper>
       {teasers.map((teaser, index) => {

```

This turns it into an actual teaser.